### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR upgrades the front-end dependencies and also adds [stylelint](https://styled-components.com/docs/tooling#stylelint)

### CODE REVIEW

File: packages/strapi-admin/admin/src/translations/en.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   "Provider": "Provider",
   "ResetPasswordToken": "Reset Password Token",
   "Role": "Role",
-  "Roles & Permissions": "Roles & Permissions",
+  "Roles & Permissions": "Roles & Permissrions",
```
Comment (soupette) [inline packages/strapi-admin/admin/src/translations/en.json:14]: `Roles & Permissions`

File: packages/strapi-plugin-content-manager/admin/src/components/DraggedField/RemoveWrapper.js [hunk 2]
```diff
@@ -1,22 +1,22 @@
 import styled from 'styled-components';
 import getColor from './utils/getColor';
 
+/* eslint-disable indent */
+
```
Comment (virginieky) [inline packages/strapi-plugin-content-manager/admin/src/components/DraggedField/RemoveWrapper.js:5]: We don't need this eslint-disable if we indent the file this way:

```
const RemoveWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  position: absolute;
  top: 0px;
  bottom: 0px;
  right: 0;
  width: 30px;
  text-align: center;
  background-color: ${({ isOverEditBlock, isOverRemove, isSelected }) => {
  return getColor(isOverRemove, isSelected, isOverEditBlock);
}};
  cursor: pointer;

  svg {
    align-self: center;
    color: #b4b6ba;
  }

  ${({ isOverRemove }) => isOverRemove && `
    path {
      fill: #f64d0a;
    }
  `};
`;
```

File: packages/strapi-plugin-content-type-builder/admin/src/components/HeaderNavLink/Wrapper.js [hunk 3]
```diff
@@ -2,18 +2,18 @@ import styled, { css } from 'styled-components';
 
 const Wrapper = styled.div`
   ${({ isActive }) => {
-    if (isActive) {
-      return css`
+  if (isActive) {
+    return css`
```
Comment (virginieky) [inline packages/strapi-plugin-content-type-builder/admin/src/components/HeaderNavLink/Wrapper.js:6]: Is `css` needed?

File: docs/3.0.0-beta.x/plugin-development/frontend-development.md [hunk 4]
```diff
@@ -60,19 +60,23 @@ Each plugin exports all its configurations in an object. This object is located
...
+| isReady                   | boolean | The app will load until this proprety is true                                                           |
+| leftMenuLinks             | array   | `[]`                                                                                                    |
+| mainComponent             | node    | The plugin's App container, setting it to null will prevent the plugin from being displayed in the menu |
+| name                      | string  | The plugin's name retrieved from the package.json.                                                      |
+| pluginLogo                | file    | The plugin's logo                                                                                       |
+| preventComponentRendering | boolean | Wheter or not display the plugin's blockerComponent instead of the main component                       |
+| settings                  | object  | Refer to ther [Plugins settings API](./frontend-settings-api.md)                                        |
+| reducers                  | object  | The plugin's redux reducers                                                                             |
```
[the remark this answers is not in the document]
Comment (soupette) [reply docs/3.0.0-beta.x/plugin-development/frontend-development.md:78]: We don't load them the same way as we directly use the `useReducer` hook so I think the description is accurate? Maybe I don't understand your comment correctly.

[the remark this answers is not in the document]
Comment (HichamELBSI) [reply docs/3.0.0-beta.x/plugin-development/frontend-development.md:78]: Alright I understand

File: packages/strapi-plugin-content-manager/admin/src/components/DraggedField/RemoveWrapper.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/components/DraggedField/RemoveWrapper.js:5]: I am not able to apply the correct indentation with prettier. Maybe you can submit a PR that fixes it?

File: docs/3.0.0-beta.x/plugin-development/frontend-development.md [hunk 5]
```diff
@@ -60,19 +60,23 @@ Each plugin exports all its configurations in an object. This object is located
...
+| trads                     | object  | The plugin's translation files                                                                          |
```
[the remark this answers is not in the document]
Comment (soupette) [reply docs/3.0.0-beta.x/plugin-development/frontend-development.md:79]: I change the table header to description.

File: docs/3.0.0-beta.x/plugin-development/frontend-development.md [hunk 6]
```diff
@@ -71,10 +71,10 @@ Here are its properties:
 | isReady                   | boolean | The app will load until this proprety is true                                                           |
 | leftMenuLinks             | array   | `[]`                                                                                                    |
```
Comment (HichamELBSI) [inline docs/3.0.0-beta.x/plugin-development/frontend-development.md:72]: If the third column is a description, you will need to update the `leftMenuLinks`, `blockerComponent`, and `blockerComponentProps` description.

File: packages/strapi-admin/admin/src/components/HeadersInput/tests/index.test.js [hunk 7]
```diff
@@ -31,7 +31,7 @@ describe('Admin | components | HeadersInput', () => {
 
     it('should match the snapshot', () => {
       const { asFragment } = render(
-        <IntlProvider locale="en">
+        <IntlProvider locale="en" textComponent="span">
```
[the remark this answers is not in the document]
Comment (soupette) [reply packages/strapi-admin/admin/src/components/HeadersInput/tests/index.test.js:34]: I agree!

File: packages/strapi-plugin-content-manager/admin/src/components/DraggedField/RemoveWrapper.js (hunk 2 above)
Comment (virginieky) [reply packages/strapi-plugin-content-manager/admin/src/components/DraggedField/RemoveWrapper.js:5]: Yes sure!

**Pull request merged**